### ISSUE TITLE
Geocoder raises "Google API error: over query limit." error if non-valid address is entered

### ISSUE THREAD

**Issue by eric-norcross** [alexreisner/geocoder#1288]
BEFORE POSTING AN ISSUE, PLEASE MAKE SURE THE PROBLEM IS NOT ADDRESSED IN THE README!

### Expected behavior
When attempting to geocode invalid input via the Google API, it should raise a `Geocoder::InvalidRequest` (or similar) error.

### Actual behavior
When attempting to geocode invalid input via the Google API, it raises a `Google API error: over query limit.` error.

### Steps to reproduce
`Geocoder.coordinates("alksdjf")` produces a `Google API error: over query limit.` error.
`Geocoder.coordinates("25 Main St, Cooperstown, NY")` behaves as expected.

### Environment info

* Geocoder version: 1.4.7
* Rails version: 5.1.4
* Ruby version: 2.5

**Comment by Frexuz** [comment]
We have this problem as well. Not using `always_raise`.
```
property.address
=> "Cheung Chau Ferry Pier, Crooks Rapid, Sai Kung, Islands, Hong Kong"
[9] pry(main)> coordinates = Geocoder.coordinates(property.address)
Google API error: over query limit.
=> nil
[10] pry(main)> coordinates = Geocoder.coordinates('asd4n2gn')
=> nil
```

The erroring address gives us a partial match -> https://www.google.com/maps/search/Cheung+Chau+Ferry+Pier,+Crooks+Rapid,+Sai+Kung,+Islands,+Hong+Kong/@22.340281,114.0185396,11z/data=!3m1!4b1
`asd4n2gn` gives us 'No results'

**Comment by eric-norcross** [comment]
@alexreisner I've been getting `over query limit` errors more frequently since posting the issue, regardless of the `always_raise` settings. I'm wondering if they are related to the upcoming changes to the [Google Maps API](https://mapsplatform.googleblog.com/2018/05/introducing-google-maps-platform.html?utm_source=em-anncmt&utm_medium=em&utm_campaign=fy18-q2-global-adhoc-email-cs-maps_none_maps_spicestandardplannobilling&utm_content=global%7Cen%7Chybr%7C%7C%7C%7C%7C%7C)

**Comment by abdelbk** [comment]
I can confirm the fact that the `over query limit` error is starting to show frequently but randomly recently. I'm pretty sure I'm not exceeding the quota.

**Comment by alexreisner** [comment]
Are you using an API key? One cause of intermittent query limit errors is a shared hosting environment or other setup in which Google can't distinguish your requests from those of other users.

**Comment by abdelbk** [comment]
I specified the API key and it works better. Thank you.

**Comment by Adilhussain003** [comment]
I have specified the API key in config but still getting over query limit error. Anyone still facing the issue?

**Comment by sandroqz** [comment]
@Adilhussain003 me too. :/

**Comment by MasashiYokota** [comment]
@Adilhussain003 @sandroqz 
I faced the same issue, but solved it.
The `query limit error` means that your application's request is denied  because your application sent too many requests per second or per day. If you want to solve this issue, you should save your usage  or increase usage limits by purchasing additional allowance for the GooleMap API or Geocoding API.

If you want to know more detail about this, please read below link.
https://developers.google.com/maps/premium/previous-licenses/articles/usage-limits

**Comment by LizPrescott** [comment]
Hi! I'm in the same boat. Our Google account is unlimited, so I don't think we're going over. I checked the Google console and didn't see so much activity that we would be going over the per-second query limit.

**Comment by rmjutras** [comment]
I'm also having this issue where I'm specifying an API key, with billing linked, but still can't geocode more than 100ish addresses without hitting this. Anyone have any insights?

**Comment by LizPrescott** [comment]
Yes, I have a few. First, this did stop happening to me when I got the latest version of geocoder (probably because the default changed to open street maps, not google anymore), so def do that if you haven't already. Second, and maybe this is obvious, but make sure you aren't making repetitive calls to the api through automation. Yes, stub out your tests, but consider seed data as well. My working theory is that this happened a lot this summer because we did a lot of db:setup because lots of onboarding summer interns and volunteers, and probably calling the api on the very same address multiple times in a day (maybe even within a minute when I was leading both of them in setting up) made google mad. It's also very likely against the Open Street Maps terms of use, so might still cause problems if it's an ongoing issue. (It isn't for us because no more interns, but I would love to hear if someone has a solution to this problem.)

**Comment by Keijun-KUMAGAI** [comment]
I have the same problem. Keep getting the error "Google API error: over query limit" where the method GeoCoder.search("address") Interestingly, in the newer account's API, it worked well. (I created a new account just for testing and check if it's because of my account or server). I want you guys to do the same thing and test it out, please.

**Comment by ankitchachada** [comment]
I am having the same issue, there is no data in my account and getting the same results.

**Comment by Mizumaki** [comment]
I have the same issue.
Any address returns nil.

```
irb(main):001:0> Geocoder.coordinates('jason statham')
Google API error: over query limit.
=> nil
irb(main):002:0> Geocoder.coordinates('神奈川県川崎市幸区大宮町１３１０ ミューザ川崎１４３')
Google API error: over query limit.
=> nil
```

**Comment by monfresh** [comment]
I am seeing this as well without even using Google. Here is my initializer:

```ruby
Geocoder.configure(
  ip_lookup: :geoip2,
  geoip2: {
    file: Rails.root.join('data', 'GeoLite2-City_20181106', 'GeoLite2-City.mmdb'),
  }
)
```
If I do `Geocoder.search('1.2.3.').first`, I see `Google API error: over query limit.` in the console, but it doesn't raise an error in the app.

**Comment by alexreisner** [comment]
@monfresh because you're entering an invalid IP address, Geocoder is treating it as a street address (hence using Google).

**Comment by monfresh** [comment]
Gotcha. That makes sense. I was trying to make sure my app could fall back to "unknown location" if the geocoder city, state, and country were all blank, and that's what came to mind. Is there a known valid IP address that would return blank for country?
